feat: auto-create logs dir if it's absent

xzmeng opened the pull request.

xzmeng committed:
feat: auto-create logs dir if it's absent

xmatthias commented on freqtrade/loggers/__init__.py:
```diff
@@ -83,6 +84,7 @@ def setup_logging(config: Config) -> None:
             handler_jd.setFormatter(Formatter("%(name)s - %(levelname)s - %(message)s"))
             logging.root.addHandler(handler_jd)
         else:
+            Path(logfile).parent.mkdir(parents=True, exist_ok=True)
```
I'm not sure this is a good way if i'm honest.

While I can see what you're trying to "protect" from (errors when given a non-existent directory) - 
i think this is made too simplistic and will need "proper" errorhandling.

Assume we're passing a subdirectory in a location we can't write - maybe by accident? (e.g. `--logfile /user_data/logs/whatever.log` - notice it's going to the root directory)? 
we've just moved the error from "FileNotFound" to "PermissionError".

xzmeng replied:
the motivation is for convenience rather than protection, we cannot prevent the user from writing the wrong path anyway.

agree that the error handling, permission need to take carefully,i did stumble into some PermissionError yesterday, but I don't have any idea ...

xmatthias replied:
well ideally, we catch the `PermissionError` and raise a (more expressive) `ConfigurationError` with a more helpful text (essentially which says "your log-directory could not be created").

the plain `permission error` is not helpful to the user as it's absolutely unclear why you'd get this error - and from experience - most non-dev-users have difficulty reading stacktraces.

xzmeng committed:
feat: add user friendly message on permission error

xmatthias committed:
feat: Exit with exception, not with exit1

this aligns to how other parts of the code work - leaving "exit" to the outermost caller.

xmatthias committed:
tests: reset permissions on tmp-path

to facilitate cleanup

xmatthias reviewed (approved):
I've made a small change to exit via OperationalException instead of exit - as i don't think we should have exit calls all over the place.

Apart from this - great enhancement :100: 
LGTM :+1:
